include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -340,6 +340,25 @@ typedef struct OrtAllocator {
    * those made during session initialization. This allows for separate memory management strategies for these allocations.
    */
   void*(ORT_API_CALL* Reserve)(struct OrtAllocator* this_, size_t size);  ///< Returns a pointer to an allocated block of `size` bytes
+
+  /**
+   * @brief Function used to get the statistics of the allocator.
+   *
+   * Return a pointer to the OrtKeyValuePairs structure that contains the statistics of the allocator.
```

Comment:
nit: Good to point out that the user should call ReleaseKeyValuePairs as they will own the returned memory.